File: app/src/main/java/com/example/secondar/RecyclerViewAdapter.java
package com.example.secondar;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>{
    private ArrayList<String> textNames;
    private ArrayList<Integer> imagesPath;
    private ArrayList<String> modelNames;
    private IFurniture furniture;

    public RecyclerViewAdapter(ArrayList<String> textNames, ArrayList<Integer> imagesPath, ArrayList<String> modelNames, IFurniture furniture) {
        this.textNames = textNames;
        this.imagesPath = imagesPath;
        this.modelNames = modelNames;
        this.furniture = furniture;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_list_item,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.imageView.setImageResource(imagesPath.get(position));
        holder.textView.setText(textNames.get(position));
        holder.imageView.setOnClickListener(view -> furniture.onClickType(position));
    }

    @Override
    public int getItemCount() {
        return imagesPath.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        ImageView imageView;
        TextView textView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.imageview);
            textView = itemView.findViewById(R.id.text);
        }
    }
}

File: app/src/main/java/com/example/secondar/RecyclerViewAdapter.kt
package com.example.secondar

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.secondar.models.Product

import java.util.ArrayList
import android.R.attr.data
import android.text.method.TextKeyListener.clear



class RecyclerViewAdapter(private val productsList: ArrayList<Product>, private val furniture: IFurniture)
    : RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.layout_list_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.imageView.setImageResource(productsList[position].imagesPath)
        holder.textView.text = productsList[position].textNames
        holder.imageView.setOnClickListener { view -> furniture.onModelItemClick(productsList[position].modelsName) }
    }

    fun updateProductList(datas: ArrayList<Product>) {
        productsList.clear()
        productsList.addAll(datas)
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return productsList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var imageView: ImageView = itemView.findViewById(R.id.imageview)
        internal var textView: TextView = itemView.findViewById(R.id.text)
    }
}